common/lib/xmodule/xmodule/capa_module.py

```diff
@@ -263,7 +266,7 @@ def has_support(self, view, functionality):
         Returns whether the given view has support for the given functionality.
         """
         if functionality == "multi_device":
-            return all(
+            return self.problem_types and all(
```

Comment:
Got it.  

1. You're returning an empty dict, not a set.  Probably doesn't matter, but the way to get an empty set is `set()`.
2. Maybe return None instead of an empty `$collection`, so you can distinguish between the two outcomes with `is not None`.